### PR TITLE
[opt](inverted index) add inlist condition handling to compound

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

1. Previously, the compound did not support the inlist condition, which could impact performance if an inverted index was created.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -1008,7 +1013,7 @@ Status SegmentIterator::_apply_inverted_index_on_column_predicate(
         }
 
         auto pred_type = pred->type();
-        if (pred_type == PredicateType::MATCH) {
+        if (pred_type == PredicateType::MATCH || pred_type == PredicateType::IN_LIST) {
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1016]: why this?

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run external

**Review by airborne12 — APPROVED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run external

**Comment by xiaokang** [comment]
run p0 10

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Pull request merged**